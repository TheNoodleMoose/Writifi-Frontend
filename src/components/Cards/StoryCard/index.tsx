import React from "react";
import styled from "styled-components";

interface StoryCardProps {
  title: String;
  author: String;
  description: String;
  createdAt: String;
}

const StoryCard = ({
  title,
  author,
  description,
  createdAt
}: StoryCardProps) => {
  return (
    <CardContainer>
      <TitleText>{title}</TitleText>
      <DescriptionText>{description}</DescriptionText>
      <DetailsContainer>
        <AuthorText>{`${author}`}</AuthorText>
        <DateText>{`@ ${createdAt}`}</DateText>
      </DetailsContainer>
    </CardContainer>
  );
};

export default StoryCard;

const CardContainer = styled.div`
  background-color: white;
  margin: 20px 0 20px 20px;
  padding: 20px;
  border-radius: 4px;
  width: 45%;
  min-height: 150px;
  min-width: 275px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  @media (max-width: 768px) {
    padding: 15px;
    margin: 20px 0;
  }
`;

const TitleText = styled.h3`
  font-size: 1.3em;
`;

const DescriptionText = styled.p`
  font-size: 1em;
`;

const DetailsContainer = styled.div`
  display: flex;

  @media (max-width: 768px) {
    width: 100%;
  }
`;

const AuthorText = styled.h3`
  margin-right: 5px;
  font-weight: 700;
`;

const DateText = styled.p`
  color: lightgray;
`;
