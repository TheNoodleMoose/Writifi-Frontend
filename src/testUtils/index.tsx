import React, { ReactElement, ReactNode } from "react";

import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

export const renderWithContext = (component: ReactElement) => {
  const history = createMemoryHistory();
  const wrapper = ({ children }: { children?: ReactNode }) => (
    <Router history={history}>{children}</Router>
  );

  const renderResults = render(component, { wrapper });
  return {
    ...renderResults,
    snapshot: () => {
      expect(renderResults.container).toMatchSnapshot();
    }
  };
};
