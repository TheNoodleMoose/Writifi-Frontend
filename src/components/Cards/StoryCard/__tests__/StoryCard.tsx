import React from "react";
import { renderWithContext } from "../../../../testUtils";

import StoryCard from "..";

it("renders correctly", () => {
  renderWithContext(<StoryCard />).snapshot();
});
