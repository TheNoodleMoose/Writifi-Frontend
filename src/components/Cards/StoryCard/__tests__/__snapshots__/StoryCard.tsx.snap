// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="sc-AxjAm dFThfg"
  >
    <div
      class="sc-AxirZ gVoMek"
    >
      <h3
        class="sc-AxiKw ccrnDg"
      >
        A cool story title
      </h3>
      <p
        class="sc-AxhCb kQwkjh"
      >
        This is a placeholder description for the test...
      </p>
      <div
        class="sc-AxhUy jZLOLz"
      >
        <h3
          class="sc-AxgMl gQjXZo"
        >
          Christian Huffman
        </h3>
        <p
          class="sc-AxheI nkKvw"
        >
          @ Some time...
        </p>
      </div>
    </div>
  </div>
</div>
`;
